src/use_file.rs

```diff
@@ -33,12 +32,15 @@ pub fn getrandom_inner(dest: &mut [MaybeUninit<u8>]) -> Result<(), Error> {
 // bytes. The file will be opened exactly once. All subsequent calls will
 // return the same file descriptor. This file descriptor is never closed.
 fn get_rng_fd() -> Result<libc::c_int, Error> {
-    static FD: AtomicUsize = AtomicUsize::new(FD_UNINIT);
+    // std::os::fd::{BorrowedFd, OwnedFd} guarantee that -1 is not a valid file descriptor.
+    const FD_UNINIT: i32 = -1;
+
+    static FD: AtomicI32 = AtomicI32::new(FD_UNINIT);
 
     fn get_fd() -> Option<libc::c_int> {
         match FD.load(Relaxed) {
             FD_UNINIT => None,
-            val => Some(val as libc::c_int),
+            val => Some(val),
```

Comment:
I think it may be worth to keep the `as` cast here to be extra safe (or use `val.try_into().map_err(|_| Error::UNEXPECTED)?`). Per `std::os::raw::c_int` [docs](https://doc.rust-lang.org/std/os/raw/type.c_int.html):

>This type will almost always be [i32](https://doc.rust-lang.org/std/primitive.i32.html), but may differ on some esoteric systems. The C standard technically only requires that this type be a signed integer that is at least the size of a [short](https://doc.rust-lang.org/std/os/raw/type.c_short.html); some systems define it as an [i16](https://doc.rust-lang.org/std/primitive.i16.html), for example.

Reply:
It is true that in theory it could be something other than `i32`. However, unless/until we require `use_file` to work on a target OS that has a different representation, there's no need to worry about that happening. This would fail to compile on such systems and there would be no silent conversion.

Reply:
Could you add a short comment about it?

Reply:
I added a comment. I also change the type of `FD_UNINIT` from `i32` to `libc::c_uint` to make it clearer.